pkgs/by-name/fo/foxglove-studio/package.nix

```
@@ -0,0 +1,130 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  dpkg,
+  glibc,
+  gcc-unwrapped,
+  autoPatchelfHook,
+  makeDesktopItem,
+  makeWrapper,
+  copyDesktopItems,
+  alsa-lib,
+  at-spi2-atk,
+  at-spi2-core,
+  atk,
+  cairo,
+  cups,
+  dbus,
+  expat,
+  gtk3,
+  libGL,
+  libX11,
+  libXcomposite,
+  libXdamage,
+  libXext,
+  libXfixes,
+  libXrandr,
+  libdrm,
+  libxcb,
+  libxkbcommon,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  udev,
+  x11basic,
+}:
+stdenv.mkDerivation rec {
+  name = "foxglove-studio-${version}";
+  version = "2.29.1";
+
+  src = fetchurl {
+    url = "https://get.foxglove.dev/desktop/v${version}/foxglove-studio-${version}-linux-amd64.deb";
+    hash = "sha256-WkyHlCTqbXikhMqSl+vkUHDe3KW9ika+SbHbn9KT7tk=";
+  };
+
+  system = "x86_64-linux";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    dpkg
+    makeWrapper
+  ];
+
+  buildInputs = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    gcc-unwrapped
+    glibc
+    gtk3
+    libGL
+    libX11
+    libXcomposite
+    libXdamage
+    libXext
+    libXfixes
+    libXrandr
+    libdrm
+    libxcb
+    libxkbcommon
+    mesa
+    nspr
+    nss
+    pango
+    udev
+    x11basic
+  ];
+
+  unpackPhase = "dpkg-deb -x $src .";
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,opt,share}
+
+    cp -r opt/Foxglove $out/opt/
+    cp -r usr/share/icons $out/share/
+    cp -r usr/share/mime $out/share/
+
+    ln -s "$out/opt/Foxglove/foxglove-studio" $out/bin/foxglove-studio
+
+    runHook postInstall
+  '';
+
+  postFixup = lib.optionalString stdenv.hostPlatform.isLinux ''
+    wrapProgram $out/bin/foxglove-studio --prefix LD_LIBRARY_PATH : ${libGL}/lib
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "foxglove-studio";
+      desktopName = "Foxglove Studio";
+      comment = "Integrated robotics visualization and debugging.";
+      exec = "foxglove-studio %U";
+      icon = "foxglove-studio";
+      categories = [ "Development" ];
+      mimeTypes = [
+        "application/octet-stream"
+        "application/zip"
+        "x-scheme-handler/foxglove"
+      ];
+    })
+  ];
+
+  meta = with lib; {
+    changelog = "https://docs.foxglove.dev/changelog/foxglove/v${version}";
+    description = "Foxglove is a visualization and observability platform for robotics.";
+    downloadPage = "https://foxglove.dev/download";
+    homepage = "https://foxglove.dev/";
+    license = licenses.unfree;
+    maintainers = with lib.maintainers; [ sascha8a ];
+    platforms = [ "x86_64-linux" ];
+  };
```

```suggestion
  meta = {
    changelog = "https://docs.foxglove.dev/changelog/foxglove/v${finalAttrs.version}";
    description = "Foxglove is a visualization and observability platform for robotics.";
    downloadPage = "https://foxglove.dev/download";
    homepage = "https://foxglove.dev/";
    license = lib.licenses.unfree;
    maintainers = with lib.maintainers; [ sascha8a ];
    platforms = [ "x86_64-linux" ];
  };
```
Avoid `with lib`, it's a large scope and it slows down eval